src/main/kotlin/controller/Controller.kt

```diff
@@ -11,38 +12,71 @@ class Controller {
         inputView: InputView,
         outputView: OutputView,
     ) {
-        val lotto = handleLottoPurchase(inputView, outputView)
+        val lotto = setNumbersOfEachTickets(inputView, outputView)
+        fillNumbersToEachTickets(lotto, inputView, outputView)
+        outputView.displayNumberOfLottoTickets(lotto)
+        outputView.displayTickets(lotto)
         val winningCombination = handleWinningCombination(inputView, outputView)
         handleResultDisplay(lotto, winningCombination, outputView)
     }
 
-    private fun handleLottoPurchase(
+    private fun setNumbersOfEachTickets(
         inputView: InputView,
         outputView: OutputView,
     ): Lotto {
-        val lotto = createLottoWithRetries(inputView)
-        outputView.displayPurchaseAmount(lotto)
-        outputView.displayNumberOfLottoTickets(lotto)
-        lotto.generateTickets()
-        outputView.displayTickets(lotto)
-        return lotto
-    }
-
-    private fun createLottoWithRetries(
-        inputView: InputView,
-        maxAttempts: Int = 3,
-    ): Lotto {
+        val maxAttempts = 3
         var lastException: IllegalArgumentException? = null
         for (i in 1..maxAttempts) {
             try {
                 val purchaseAmount = inputView.getPurchaseAmount()
-                return Lotto(purchaseAmount)
+                val lotto = Lotto(purchaseAmount)
+                outputView.displayPurchaseAmount(lotto)
+                val manualCount = inputView.getNumberOfManualTickets()
+                lotto.numberOfManualTickets = manualCount
+                outputView.displaySingleNumber(lotto.numberOfManualTickets)
+                lotto.numberOfAutomaticTickets = lotto.numberOfTotalTickets - manualCount
+                return lotto
             } catch (e: IllegalArgumentException) {
-                println("${e.message}")
+                println(e.message ?: "Unknown error")
                 lastException = e
             }
         }
-        throw lastException!!
+        throw lastException ?: IllegalArgumentException("Unknown error")
```

Comment:
Instead of keeping the exception for later, why don't we just :
```Kotlin
    throw IllegalArgumentException("Too many failed attempts. Please restart the program.")
```

Also, I think we can utilize `repeat(maxAttempts)` here.